### ISSUE TITLE
Default value not used ?

### ISSUE THREAD

**Issue by fletort** [github/local-action#111]
I think that default value seems not to be used for input, right ?
It should be possible to add this ?
It should be interresting to can test these default value behaviour from the cli.

**Comment by ncalteen** [comment]
At the moment, no default values are used. I can definitely set these!

For now, I will set them to assume that the user is running `local-action` from the directory containing the action repository, so:

```bash
local-action run . src/main.ts .env
```

**Comment by ncalteen** [comment]
Hmm....actually on second thought I think I might hold off on this. The issue is the tool doesn't currently know the type of action being run (JavaScript or TypeScript...or hopefully eventually container/composite).

**Comment by theoephraim** [comment]
This was very unexpected - and it means we must explicitly pass in all the defaults again.
Regardless of the action type - I think it should respect those defaults defined in the action.yaml file.

**Comment by ncalteen** [comment]
Apologies, I think I misunderstood the original intent of the request. I was assuming this meant the defaults for the `local-action` command itself, not the default inputs in the `action.yml` file. I can certainly add that in!

**Comment by theoephraim** [comment]
Awesome - thanks!

With this now working, I'm not sure if having a .env file is always necessary. Might want to remove that requirement?

**Comment by ncalteen** [comment]
I would say yes but there are a bunch of other configuration flags in there, not including inputs.

**Comment by theoephraim** [comment]
Totally - but they are not _required_, and can be passed in via actual env vars. Currently I'm only using inputs in mine, so once this lands, I'll have the requirement of creating an empty .env file which feels a bit weird.

**Comment by theoephraim** [comment]
Also! Getting an error now `::error::EnvMeta.inputs[name].default.trim is not a function`

**Comment by theoephraim** [comment]
ok - confirmed that the issue is that the default value coming through is a boolean
The yaml treats`default: true` and `default: 'true'` differently.  The TS type and your related test assume it's always coming through as a string.

**Comment by ncalteen** [comment]
Can we open a separate issue to track this?

FYI the [GitHub YAML schema](https://json.schemastore.org/github-action.json) expects a `string` value. This is also documented in the [action syntax documentation](https://docs.github.com/en/actions/sharing-automations/creating-actions/metadata-syntax-for-github-actions#inputsinput_iddescription).
